### ISSUE TITLE
Why is_valid is False? 

### ISSUE THREAD

**Issue by Coke-das-JHL** [Xilinx/brevitas#556]
from brevitas.quant import Int8ActPerTensorFixedPoint as ActQuant 

class MyBinaryQuantizer(ExtendedInjector):
    tensor_quant = BinaryQuant
    scaling_impl=ParameterScaling
    scaling_init=1.
    signed = True

class MyBinaryWeightQuantizer(MyBinaryQuantizer):
    proxy_class = WeightQuantProxyFromInjector

class MySignedBinaryActQuantizer(MyBinaryQuantizer):
    proxy_class = ActQuantProxyFromInjector

self.PFN_act_bitwidth = 16 
in_channel = 9
out_channel = 64

self.quant_inp = qnn.QuantIdentity(act_quant=ActQuant, bit_width=self.PFN_act_bitwidth, return_quant_tensor=True)
self.conv      = qnn.QuantConv1d(in_channel, out_channel, 1, bias=False, weight_quant=MyBinaryWeightQuantizer, return_quant_tensor=True)

OUTPUT ( Inference) : 
![image](https://user-images.githubusercontent.com/68185836/225842850-5745d632-c62b-4afe-88e8-140887b66fd9.png)

is_int = torch.isclose(pre_round_int_value, rounded_int_value, atol=IS_VALID_ATOL).all()  # False

==============================================================================

It was confirmed that False occurred in is_int.

How can I solve the problem ? 

In the inference stage, even if is_valid is False, shouldn't the Tensor(rounded_int_value = torch.round(pre_round_int_value)) containing the error be delivered to next layer?

Also, can you tell me the impact of is_valid on HW implementation ?

**Comment by Giuseppe5** [comment]
Hi,
Thanks for your question.

I am not I understand your script and what you're trying to do.
Could you please it share it with Colab or similar?
Also, if you could provide more information about what Brevitas/Pytorch version you're using, that would be very helpful as well.

Many thanks,
Giuseppe

**Comment by Coke-das-JHL** [comment]
First of all, I'm sorry.

I am checking the 'is_valid' value of Activation (QuantTensor) in the inference process, and 'is_valid' has a false value.

Can you tell me why this is happening? How can I solve it?

When is_valid is False, are there any considerations for hardware implementation?

Thank you. 

( Brevitas/0.7.1 )

**Comment by Giuseppe5** [comment]
I am not sure how to replicate your issue.

Generally speaking, if your model/quant_modules are in training mode, the output QuantTensors might be not valid, due how to the scale factors are computed.

If you put your model/quant_modules, all your quant tensors should be valid, which is also (generally  speaking) the state that you want your model to be in before moving to hardware implementation.

**Comment by Coke-das-JHL** [comment]
I want to do a BWN simulation. 

Int8ActPerTensorFloat Quantizer is used for activation, and the following Quantizer is used for weight. 

In this case, the is_valid of the output of the convolution layer is False. Can you give me some advice?

```python
class MyBinaryQuantizer(ExtendedInjector):
    tensor_quant = BinaryQuant
    scaling_impl=ParameterScaling
    scaling_init=1.
    signed = True

class MyBinaryWeightQuantizer(MyBinaryQuantizer):
    proxy_class = WeightQuantProxyFromInjector
```
Part of the model
```python 

self.PFN_act_bitwidth = 16

self.quant_inp = qnn.QuantIdentity(act_quant=ActQuant, bit_width=self.PFN_act_bitwidth, return_quant_tensor=True)
self.conv      = qnn.QuantConv1d(in_channel, out_channel, 1, bias=False, weight_quant=MyBinaryWeightQuantizer, return_quant_tensor=True)
self.relu      = qnn.QuantReLU(output_quant=ActQuant, bit_width=self.PFN_act_bitwidth, return_quant_tensor=True)

# forward part
 features = self.quant_inp(features)
 features = self.conv(features)
 features = self.relu(features)
```
